common/tqid/task_queue_id.go

```diff
@@ -357,6 +360,81 @@ func (p *NormalPartition) RoutingKey() string {
 	return fmt.Sprintf("%s:%s:%d", p.NamespaceId(), p.RpcName(), p.TaskType())
 }
 
+// ValidateTaskQueue validates a TaskQueue object.
+// It checks the TaskQueue's name for emptiness, length, UTF-8 validity, and whitespace.
+// For sticky queues, it also validates the NormalName.
+// If the name is empty and defaultVal is provided, it sets the name to defaultVal.
+// If the Kind is unspecified, it sets it to NORMAL.
+//
+// Parameters:
+//   - taskQueue: The TaskQueue to validate. If nil, returns an error.
+//   - defaultVal: Default name to use if taskQueue name is empty.
+//   - maxIDLengthLimit: Maximum allowed length for the TaskQueue name.
+//
+// Returns an error if validation fails, nil otherwise.
+func ValidateTaskQueue(
+	taskQueue *taskqueuepb.TaskQueue,
+	defaultVal string,
+	maxIDLengthLimit int,
+) error {
+	if taskQueue == nil {
+		return serviceerror.NewInvalidArgument("TaskQueue is not set.")
+	}
+
+	enums.SetDefaultTaskQueueKind(&taskQueue.Kind)
+
+	if taskQueue.GetName() == "" {
+		if defaultVal == "" {
+			return serviceerror.NewInvalidArgument("Missing task queue name.")
+		}
+		taskQueue.Name = defaultVal
+	}
+
+	if err := ValidateTaskQueueName(taskQueue.GetName(), maxIDLengthLimit); err != nil {
+		return err
+	}
+
+	if strings.HasPrefix(taskQueue.GetName(), reservedTaskQueuePrefix) {
+		return serviceerror.NewInvalidArgument(fmt.Sprintf("Task queue name cannot start with reserved prefix %v.", reservedTaskQueuePrefix))
+	}
+
+	if taskQueue.GetKind() == enumspb.TASK_QUEUE_KIND_STICKY {
+		if err := ValidateTaskQueueName(taskQueue.GetNormalName(), maxIDLengthLimit); err != nil {
```

Comment:
It looks like previously the only extra check we were doing for sticky queues was `common.ValidateUTF8String(taskQueue.GetNormalName()` so I just want to be very sure that adding extra validation here won't break anything.

Reply:
Your assessment is correct, and that looked dubious to me. I cannot think of why we wouldn't want the additional validation for the normal name.